wcd2reshare.py

```diff
@@ -0,0 +1,88 @@
+import urllib.parse
+
+baseURL = "https://borrowdirect.reshare.indexdata.com/Search/Results?"
+
+
+def query_formatter(args: dict) -> str:
+    """transforms a dictionary of OpenURL request params to a string of search
+    params formatted for BorrowDirect / ReShare vuFind UI"""
+
+    # logic is:
+    # if isbn
+    #   use isbn and nothing else
+    # else if any of the 4 titles exist:
+    #   use the title that exits in this preferential order
+    #      title, btitle, ctitle, jtitle
+    #      nested condition of if aulast include that as well
+
+    searchString = ""
+    isbn = args.get("rft.isbn")
+    if isbn:
+        searchString = [("type", "ISN"), ("lookfor", isbn)]
+        return urllib.parse.urlencode(searchString)
+
+    aulast = args.get("rft.aulast")
+    title = args.get("rft.title")
+    if title:
+        searchString = [("type", "title"), ("lookfor", title)]
+        if aulast:
+            searchString = [
+                ("join", "AND"),
+                ("type0[]", "title"),
+                ("lookfor0[]", title),
+                ("type0[]", "author"),
+                ("lookfor0[]", aulast),
+            ]
+        return urllib.parse.urlencode(searchString)
```

Comment:
This code block is repeated 4 times so it should be a function with its own unit tests because all that's really changing is which title is being retrieved, right? Also, the way it's written now, it's setting the `searchString` variable after getting the title and then possibly overwriting that variable if `aulast` exist. Could this be written so `searchString` is only set once?